client/landing/stepper/hooks/use-checkout.ts

```diff
@@ -33,14 +34,19 @@ const useCheckout = () => {
 		setSignupCompleteFlowName( flowName );
 		setSignupCompleteStepName( stepName );
 
-		let checkoutUrl = `/checkout/${ encodeURIComponent( siteSlug ) }`;
-		if ( plan ) {
-			checkoutUrl += `/${ plan }`;
-		}
+		// If the plan is not provided, we might have added plan to the cart so we just go to the checkout page directly
+		if ( ! plan ) {
+			let checkoutUrl = `/checkout/${ encodeURIComponent( siteSlug ) }`;
+			if ( plan ) {
```

Comment:
Is this if-statement dead code? 🤔

Reply:
ooops...I forgot to update the existing code 🙈